src/Uno.UI/UI/Xaml/VisualStateGroup.cs

```diff
@@ -348,11 +348,14 @@ void ApplyTargetStateSetters()
 
 					// This block is a manual enumeration to avoid the foreach pattern
 					// See https://github.com/dotnet/runtime/issues/56309 for details
-					var settersEnumerator = target.setters.OfType<Setter>().GetEnumerator();
+					var settersEnumerator = target.setters.GetEnumerator();
 
 					while (settersEnumerator.MoveNext())
 					{
-						settersEnumerator.Current.ApplyValue(element);
+						if (settersEnumerator.Current is Setter setter)
+						{
+							setter.ApplyValue(element);
+						}
 					}
```

Comment:
I think this pattern is not relevant anymore. It was when Wasm were not handling try/catch and it was causing a lot of overhead.

Reply:
In this case, I'd love to do a wide change to switch from `MoveNext` to `foreach` over the whole codebase. Let's track this in a new issue

Reply:
#17409